src/test/http_openai_handler_test.cpp

```diff
@@ -275,6 +275,41 @@ TEST_F(HttpOpenAIHandlerParsingTest, ParsingMessagesSucceeds) {
     EXPECT_EQ(json, std::string("{\"model\":\"llama\",\"messages\":[{\"role\":\"user\",\"content\":\"What is in this image?\"}]}"));
 }
 
+TEST_F(HttpOpenAIHandlerParsingTest, ParsingMessagesSucceedsUrl) {
```

Comment:
there is no test for downloading files with multipart and size >20MB. A functional tests should be added. If not in the PR, then later. Confirm at least manually now.
There is also no test for pulling over https using public and private certificate.